### PR TITLE
runtime: Remove static loader APIs

### PR DESCRIPTION
**Pull request by backspace**
This removes the concept of the global loader, which necessitated the updating of various interfaces to accept a loader and an explicit call to `shimExternals` when a loader is created. It should fix the test leakage seen in #467.

### CODE REVIEW

File: packages/base/card-api.gts [hunk 1]
```diff
@@ -1266,6 +1272,12 @@ class LinksToMany<FieldT extends CardConstructor>
       (ref) => new URL(ref, instance.id ?? relativeTo).href // new instances may not yet have an ID, in that case fallback to the relativeTo
     );
     let loader = Loader.getLoaderFor(createFromSerialized);
+
+    if (!loader) {
+      debugger;
```
Comment (habdelra) [inline packages/base/card-api.gts]: oops!

File: packages/base/card-api.gts [hunk 2]
```diff
@@ -936,6 +936,12 @@ class LinksTo<CardT extends CardConstructor> implements Field<CardT> {
       instance.id ?? relativeTo // new instances may not yet have an ID, in that case fallback to the relativeTo
     ).href;
     let loader = Loader.getLoaderFor(createFromSerialized);
+
+    if (!loader) {
+      debugger;
```
Comment (habdelra) [inline packages/base/card-api.gts]: oops!

File: packages/base/card-api.gts [hunk 3]
```diff
@@ -2095,6 +2108,12 @@ async function cardClassFromResource<CardT extends CardBaseConstructor>(
   }
   if (resource && !isEqual(resource.meta.adoptsFrom, cardIdentity)) {
     let loader = Loader.getLoaderFor(fallback);
+
+    if (!loader) {
+      debugger;
```
Comment (habdelra) [inline packages/base/card-api.gts]: another debugger

File: packages/base/catalog-entry.gts [hunk 4]
```diff
@@ -22,7 +22,17 @@ export class CatalogEntry extends Card {
   @field ref = contains(CardRefCard);
   @field isPrimitive = contains(BooleanCard, {
     computeVia: async function (this: CatalogEntry) {
+      let loader = Loader.getLoaderFor(Object.getPrototypeOf(this).constructor);
+
+      if (!loader) {
+        debugger;
```
Comment (habdelra) [inline packages/base/catalog-entry.gts]: remove debugger

File: packages/base/room.gts [hunk 5]
```diff
@@ -363,6 +364,15 @@ export class RoomCard extends Card {
     // tell the card runtime that this field is being used
     isUsed: true,
     computeVia: async function (this: RoomCard) {
+      let loader = Loader.getLoaderFor(Object.getPrototypeOf(this).constructor);
+
+      if (!loader) {
+        debugger;
```
Comment (habdelra) [inline packages/base/room.gts]: remove debugger

**Review by habdelra — APPROVED** [review]
you have a couple left over `debugger`s, but otherwise looks good

**Comment by backspace** [comment]
> you have a couple left over `debugger`s, but otherwise looks good

removed, thanks! The `throw`s were needed but I should have taken the `debugger`s out 🤓

**Pull request merged**